pingora-core/src/connectors/mod.rs

```diff
@@ -201,7 +201,7 @@ impl TransportConnector {
                         }
                     }
                     Err(_) => {
-                        error!("failed to acquire reusable stream");
+                        debug!("failed to acquire reusable stream");
```

Comment:
Is it possible that the connection is present, but we are still unable to acquire the stream due to system level issues? Thinking out loud, if having it as an 'warn' instead, helps for those scenarios

Reply:
well, possible other real error scenarios is exactly why I raised it as a question first on pingora's issue board https://github.com/cloudflare/pingora/issues/201. To be fair though, even in the case where it is some system level error, it really should be detected differently and logged differently. 
The way it is in its current state probably doesn't serve anyone since you can't tell exactly what happened anyway based on that ERROR log line. All it does now in my application at least is fill logs up with multiple ERRORs even with no actual problems.